File: HSTS/src/main/java/com/example/hsts/utils/ActivityUtils.java
package com.example.hsts.utils;

import android.os.Build;

import androidx.appcompat.app.AppCompatActivity;

public class ActivityUtils {

    public static void finishActivity(AppCompatActivity activity, boolean isFinish) {
        if (!isFinish) return;
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
            if (!activity.isDestroyed()) {
                activity.finish();
            }
        } else {
            activity.finish();
        }
    }
}

File: HST/src/main/java/com/hst/utils/ActivityUtils.kt
package com.hst.utils

import androidx.appcompat.app.AppCompatActivity
import android.os.Build

object ActivityUtils {
    fun finishActivity(activity: AppCompatActivity, isFinish: Boolean) {
        if (!isFinish) return
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
            if (!activity.isDestroyed) {
                activity.finish()
            }
        } else {
            activity.finish()
        }
    }
}